### ISSUE TITLE
When clicking a link that doesn't have a transition, scroll position isn't reset

### ISSUE THREAD

**Issue by TylerBarnes** [TylerBarnes/gatsby-plugin-transition-link#21]

**Comment by thundernixon** [comment]
Hi @TylerBarnes, is this issue related to, or distinct from https://github.com/gatsbyjs/gatsby/issues/7454?

On my demo site, I have an issue where the pagination links at the bottom of posts will lead to the bottom of other posts, rather than the top as I expect. 

![link-scroll-oddity](https://user-images.githubusercontent.com/7355414/52905781-1fbdb200-320d-11e9-8fc7-faabbca9b5ba.gif)

I've also reported it in the issue linked above (gatsby 7454), but I wonder if there's a chance that this issue (TL 21) has anything to do with it (or if you know what I might be doing incorrectly that could be causing this).

**Test Repo:** https://github.com/thundernixon/gatsby-link-scroll-test

**Deployed at:** https://zen-brown-8c4cd6.netlify.com/

**Comment by TylerBarnes** [comment]
Hey again @thundernixon ! I cloned your repo to test it out. The reason is that you're mixing transition link and gatbsy link. [`shouldUpdateScroll`](https://www.gatsbyjs.org/docs/browser-apis/#shouldUpdateScroll) is disabled in transition link as the timing of when to scroll to the top needs to be handled by the transition. I cloned your repo and switched your gatsby link import to a transition link import and it works properly.

In pagination.js switch: `import { Link } from 'gatsby'` for `import Link from "gatsby-plugin-transition-link";` and it will work.

I'll likely need to add a generic event listener and check if a regular gatsby link is being clicked and update the scroll position from there. For now I'll add a note to the docs to not mix gatsby link and transitionlink.
